src/plugins/data_source_management/public/components/create_edit_data_source_wizard/create_edit_data_source_wizard.tsx

```diff
@@ -147,20 +149,30 @@ export class CreateEditDataSourceWizard extends React.Component<
       title: [],
       description: [],
       endpoint: [],
+      credential: [],
     };
     const formErrorMessages: string[] = [];
+    /* Title validation */
     if (!this.state.dataSourceTitle) {
       validationByField.title.push('Title should not be empty');
       formErrorMessages.push('Title should not be empty');
     }
+    /* Description Validation */
     if (!this.state.dataSourceDescription) {
       validationByField.description.push('Description should not be empty');
       formErrorMessages.push('Description should not be empty');
     }
-    if (!this.state.endpoint) {
-      validationByField.endpoint.push('Endpoint should not be empty');
-      formErrorMessages.push('Endpoint should not be empty');
+    /* Endpoint Validation */
+    if (!isValidUrl(this.state.endpoint)) {
+      validationByField.endpoint.push('Endpoint is not valid');
+      formErrorMessages.push('Endpoint is not valid');
     }
+    /* Credential Validation */
+    if (!this.state.noAuthentication && !this.state.selectedCredentials?.length) {
+      validationByField.credential.push('Please associate a credential');
+      formErrorMessages.push('Please associate a credential');
+    }
```

Comment:
Can we also mention something about noAuth as another option in the validation message?

Reply:
if user selects noAuth checkbox, they will not see this message.

I prefer having that info about noAuth in helper text, rather than a validation message